hummingbot/strategy/__utils__/trailing_indicators/trading_intensity.pyx

```diff
@@ -46,43 +47,81 @@ cdef class TradingIntensityIndicator:
 
     @property
     def is_sampling_buffer_full(self) -> bool:
-        return len(self._trade_samples) == self._sampling_length
+        return len(self._trade_samples.keys()) == self._sampling_length
 
     @property
     def is_sampling_buffer_changed(self) -> bool:
-        is_changed = self._samples_length != len(self._trade_samples)
-        self._samples_length = len(self._trade_samples)
+        is_changed = self._samples_length != len(self._trade_samples.keys())
+        self._samples_length = len(self._trade_samples.keys())
         return is_changed
 
     @property
     def sampling_length(self) -> int:
         return self._sampling_length
 
+    # A helper method to be used in unit tests
     @property
-    def last_price(self) -> float:
-        return self._last_price
+    def last_quotes(self) -> list:
+        return self._last_quotes
+
+    # A helper method to be used in unit tests
+    @last_quotes.setter
+    def last_quotes(self, value):
+        self._last_quotes = value
+
+    # A helper method to be used in unit tests
+    def calculate(self, timestamp):
+        self.c_calculate(timestamp)
```

Comment:
```suggestion
    @property
    def last_price(self) -> float:
        """A helper method to be used in unit tests."""
        return self._last_price
    def last_quotes(self) -> list:
        return self._last_quotes

    @last_quotes.setter
    def last_quotes(self, value):
        """A helper method to be used in unit tests."""
        self._last_quotes = value

    def calculate(self, timestamp):
        """A helper method to be used in unit tests."""
        self.c_calculate(timestamp)
```

Reply:
There's no `_last_price`. Anyway I got the idea, I'll change the comments.